lib/util/state.js

```diff
@@ -106,75 +106,116 @@ export function getActiveItineraries (otpState) {
-          const aCost = calculateItineraryCost(a)
-          const bCost = calculateItineraryCost(b)
+          const aCost = calculateItineraryCost(a, config)
+          const bCost = calculateItineraryCost(b, config)
           if (direction === 'ASC') return aCost - bCost
           else return bCost - aCost
       }
     })
 }
 
+// Default constants for calculating itinerary "cost", i.e., how preferential a
+// particular itinerary is based on factors like wait time, total fare, drive
+// time, etc.
+const FARE_FACTOR = 0.5
+const DURATION_FACTOR = 0.25
+const WALK_RELUCTANCE = 0.1
+const WAIT_RELUCTANCE = 0.1
+const DRIVE_RELUCTANCE = 2
+const TRANSFER_RELUCTANCE = 0.9
+
 /**
  * This calculates the "cost" (not the monetary cost, but the cost according to
  * multiple factors like duration, total fare, and walking distance) for a
  * particular itinerary, for use in sorting itineraries.
  * FIXME: Do major testing to get this right.
  */
-function calculateItineraryCost (itinerary) {
-  // FIXME: Add drive reluctance factor?
-  // FIXME: Make these configurable.
-  const FARE_COST_FACTOR = 0.5
-  const DURATION_COST_FACTOR = 0.25
-  const WALK_COST_FACTOR = 0.1
-  const WAIT_COST_FACTOR = 0.1
-  const TRANSFER_COST_FACTOR = 0.9
-  return getTotalFare(itinerary) * FARE_COST_FACTOR +
-    itinerary.duration * DURATION_COST_FACTOR +
-    itinerary.walkDistance * WALK_COST_FACTOR +
-    itinerary.waitingTime * WAIT_COST_FACTOR +
-    itinerary.transfers * TRANSFER_COST_FACTOR
+function calculateItineraryCost (itinerary, config = {}) {
+  const weights = {
+    walkReluctance: WALK_RELUCTANCE,
+    driveReluctance: DRIVE_RELUCTANCE,
+    durationFactor: DURATION_FACTOR,
+    fareFactor: FARE_FACTOR,
+    waitReluctance: WAIT_RELUCTANCE,
+    transferReluctance: TRANSFER_RELUCTANCE
+  }
+  // If config contains values to override defaults, apply those.
+  const configWeights = config.itinerary && config.itinerary.weights
+  if (configWeights) Object.assign(weights, configWeights)
+  return getTotalFare(itinerary, config) * weights.fareFactor +
+    itinerary.duration * weights.durationFactor +
+    itinerary.walkDistance * weights.walkReluctance +
+    getDriveTime(itinerary) * weights.driveReluctance +
+    itinerary.waitingTime * weights.waitReluctance +
+    itinerary.transfers * weights.transferReluctance
+}
+
+/**
+ * Get total drive time (i.e., total duration for legs with mode=CAR) for an
+ * itinerary.
+ */
+function getDriveTime (itinerary) {
+  if (!itinerary) return 0
+  let driveTime = 0
+  itinerary.legs.forEach(leg => {
+    if (leg.mode === 'CAR') driveTime += leg.duration
+  })
+  return driveTime
 }
 
+// $2 per trip? This is a made up number.
+const DEFAULT_BIKESHARE_FARE = 2 * 100
+// $2 for 3 hours of parking?
+const DEFAULT_PARKING_COST = 3 * 2.00 * 100
+// FL per diem rate: https://www.flcourts.org/content/download/219314/1981830/TravelInformation.pdf
+const DEFAULT_DRIVING_COST_PER_MILE = 0.445 * 100
```

Comment:
These could go in the constants file

Reply:
If they are not used anywhere but this file, is there really any benefit to moving them there?